src/Apps/Order/Components/PaymentPicker.tsx

```diff
@@ -197,6 +201,22 @@ export class PaymentPicker extends React.Component<
 
     const userHasExistingCards = creditCards.edges.length > 0
 
+    const orderCard = this.props.order.creditCard
+
+    const creditCardsArray: any = []
+
+    creditCards.edges.map(e => {
+      creditCardsArray.push(e.node)
+    })
+
+    if (
+      orderCard != null &&
+      creditCardsArray.filter(card => card.id === orderCard.id).length === 0 &&
```

Comment:
You can use the new array method `some` for this :)

```suggestion
      creditCardsArray.some(card => card.id === orderCard.id) &&
```

Reply:
For some reason some isn't working for me here. We can either merge as is or pair on this on Monday if you're up for it.

Reply:
:grimacing: That's my bad, in the suggestion it should have been negated, i.e.

```ts
      !creditCardsArray.some(card => card.id === orderCard.id) &&
```

But it's no biggie feel free to keep as-is.